demos/twa-offline-first/src/main/java/com/google/androidbrowserhelper/demos/twa_offline_first/OfflineFirstTWALauncherActivity.java

```diff
@@ -0,0 +1,48 @@
+package com.google.androidbrowserhelper.demos.twa_offline_first;
+
+import com.google.androidbrowserhelper.trusted.LauncherActivity;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+public class OfflineFirstTWALauncherActivity extends LauncherActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.displayScreen();
+    }
+
+    @Override
+    protected boolean shouldLaunchImmediately() {
+        return false;
+    }
+
+    private void displayScreen() {
+        if(isOnline()) {
+            launchTwa();
+        } else {
+            launchCustomOfflineScreen();
+        }
+    }
+
+    private void launchCustomOfflineScreen() {
```

Comment:
nit: Launch makes me thing we're launching an Activity, but we're not actually launching anything here, and just setting the content of the currentActivity. Maybe rename this to `renderOfflineFallback()` or something similar?
```suggestion
    private void renderOfflineFallback() {
```